pkg/provisioner/controller.go

```diff
@@ -638,3 +648,62 @@ func (c *obcController) objectBucketForClaimKey(key string) (*v1alpha1.ObjectBuc
 	}
 	return ob, nil
 }
+
+func updateSupported(old, new *v1alpha1.ObjectBucketClaim) bool {
+	// The only field supported for update is obc.spec.additionalConfig
+	if reflect.DeepEqual(new.Spec, old.Spec) {
```

Comment:
Is it necessary to compare the whole spec if the only supported field is additionalConfig?  Why not instead check if additionalConfig has changed?

Reply:
One of the goals is to log an error or warning if the user attempts to change anything in the obc other than `.spec.additionalConfig`.  I guess DeepEqual handles map compares in cases where they keys may be in different order, right?

Reply:
Nevermind, I believe it's just to short circuit the call if nothing call a non-spec field was changed.  Makes sense to me.